tests/Common/TriggersTest.php

```diff
@@ -543,33 +564,77 @@ public function testDeleteTriggerWithNonMasterToken()
+        $anotherClientWithAnotherToken = $this->getClient([
+            'url' => STORAGE_API_URL,
+            'token' => $anotherToken['token'],
+        ]);
 
-        $this->expectException(ClientException::class);
-        $this->expectExceptionMessage(sprintf('Trigger with id [%d] was not found', $trigger['id']));
-        $clientWithoutAdminToken->getTrigger((int) $trigger['id']);
+        // trigger can be deleted by non-master token with right privilege
+        $anotherClientWithAnotherToken->deleteTrigger((int) $trigger['id']);
+        $triggerList = $clientWithoutAdminToken->listTriggers([$table1]);
+        $this->assertNotContainsTriggerId($trigger['id'], $triggerList);
+        $this->assertCount(1, $triggerList);
 
         // master token can delete it anyway
         $this->_client->deleteTrigger((int) $trigger2['id']);
+        $triggerList = $clientWithoutAdminToken->listTriggers([$table1]);
+        $this->assertNotContainsTriggerId($trigger2['id'], $triggerList);
 
-        $this->expectException(ClientException::class);
-        $this->expectExceptionMessage(sprintf('Trigger with id [%d] was not found', $trigger2['id']));
-        $this->_client->getTrigger((int) $trigger2['id']);
+        $this->assertCount(0, $triggerList);
+    }
+
+    /**
+     * @dataProvider tokenCreateOptionsProvider
+     * @return void
+     */
+    public function testDeleteTriggerCreatedByNonMasterToken(TokenCreateOptions $optionForToken)
+    {
+        $table1 = $this->createTableWithRandomData("watched-1");
+
+        $optionsForTokenRunWith = (new TokenCreateOptions())
+            ->addBucketPermission($this->getTestBucketId(), TokenAbstractOptions::BUCKET_PERMISSION_READ);
+
+        $tokenRunWith = $this->tokens->createToken($optionsForTokenRunWith);
+        $newNonAdminToken = $this->tokens->createToken($optionForToken);
+        $clientWithoutAdminToken = $this->getClient(['url' => STORAGE_API_URL, 'token' => $newNonAdminToken['token']]);
+
+        $trigger = $clientWithoutAdminToken->createTrigger([
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriodMinutes' => 10,
+            'runWithTokenId' => $tokenRunWith['id'],
+            'tableIds' => [$table1],
+        ]);
+
+        // trigger can be deleted by non-master token with right privilege
+        $clientWithoutAdminToken->deleteTrigger((int) $trigger['id']);
+        $triggerList = $clientWithoutAdminToken->listTriggers([$table1]);
+        $this->assertNotContainsTriggerId($trigger['id'], $triggerList);
+        $this->assertCount(0, $triggerList);
+    }
+
+    /**
+     * @param int $triggerId
+     * @param array $triggerList
+     * @return void
+     */
+    private function assertNotContainsTriggerId($triggerId, $triggerList)
+    {
+        $this->assertFalse(in_array($triggerId, array_map(function ($trigger) {
+            return $trigger['id'];
+        }, $triggerList)));
```

Comment:
AssertFalse má nevýhodu, že ti neřekne nic o tom, co bylo v tom poli. Je lepší assertovat že "není v poli"...

Reply:
```
Failed asserting that true is false.
 /code/tests/Common/TriggersTest.php:637
 /code/tests/Common/TriggersTest.php:526
 /code/vendor/bshaffer/phpunit-retry-annotations/src/RetryTrait.php:33

```

Reply:
```suggestion
        $this->assertNotContains($triggerId, array_map(function ($trigger) {
            return $trigger['id'];
        }, $triggerList));
```

Ale koukám, že to taky nevrací co v tom poli teda bylo :/

Reply:
Ale furt mi to přijde jako čitelnější error :)